### PR TITLE
UX-794, UX-795/Added repository and app catalog page

### PR DESCRIPTION
**Pull request by knnguy**
Repo List

Update Repo

Add Repo

Add Repo 2

App Catalog

Deploy Page

Screen Shot 2021-02-09 at 6 06 24 PM

Screen Shot 2021-02-09 at 6 06 44 PM

Screen Shot 2021-02-09 at 6 06 58 PM

Screen Shot 2021-02-09 at 6 07 15 PM

Screen Shot 2021-02-09 at 6 07 36 PM

### CODE REVIEW

**Comment by knnguy** [comment]
Not ready for review yet. I still need to go clean up the code

File: src/api-client/Helm.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      url,
+      body,
+      options: {
+        clsName: this.getClassName(),
+        mthdName: 'updateAllRepositories',
+      },
+    })
+  }
+
+  addRepoToCluster = async (repoName, clusterId) => {
```
Comment (jruddell) [inline src/api-client/Helm.ts:89]: isn't this name backwards? shouldn't it be `addClusterToRepo`?

File: src/api-client/Helm.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    return this.client.basicDelete({
+      url,
+      options: {
+        clsName: this.getClassName(),
+        mthdName: 'deleteRepoFromCluster',
+      },
+    })
+  }
+
+  getRepositoriesForCluster = async (clusterId) => {
```
Comment (jruddell) [inline src/api-client/Helm.ts:136]: we should be consistent, either call it `Repos` or `Repositories`

File: src/app/core/components/cardTable/card-table.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      return sortBy === 'desc' ? sortedData.reverse() : sortedData
+    },
+    [showSortOption, sortTarget, sortBy],
+  )
+
+  const filteredData = compose(filter, filterBySearch, sort)(data)
+
+  return (
+    <Progress overlay loading={loading} renderContentOnMount>
+      <Grid container justify="center">
```
Comment (jruddell) [inline src/app/core/components/cardTable/card-table.tsx:91]: do we need to use grid?

Comment (knnguy) [reply src/app/core/components/cardTable/card-table.tsx:91]: No, I can try to use CSS grids, but is there a reason why we don't want to use the MaterialUI grid?

**Review by jruddell — APPROVED** [review]

**Pull request merged**